sql/core/src/test/scala/org/apache/spark/sql/execution/local/LocalNodeTest.scala

```diff
@@ -0,0 +1,189 @@
+      expectedAnswer: Seq[Row],
+      sortAnswers: Boolean = true): Unit = {
+    doCheckAnswer(
+      left :: right :: Nil,
+      nodes => nodeFunction(nodes(0), nodes(1)),
+      expectedAnswer,
+      sortAnswers)
+  }
+
+  /**
+   * Runs the `LocalNode`s and makes sure the answer matches the expected result.
+   * @param input the input data to be used.
+   * @param nodeFunction a function which accepts a sequence of input `LocalNode`s and uses them to
+   *                     instantiate the local physical operator that's being tested.
+   * @param expectedAnswer the expected result in a [[Seq]] of [[Row]]s.
+   * @param sortAnswers if true, the answers will be sorted by their toString representations prior
+   *                    to being compared.
+   */
+  protected def doCheckAnswer(
+    input: Seq[DataFrame],
+    nodeFunction: Seq[LocalNode] => LocalNode,
+    expectedAnswer: Seq[Row],
+    sortAnswers: Boolean = true): Unit = {
+    LocalNodeTest.checkAnswer(
+      input.map(dataFrameToSeqScanNode), nodeFunction, expectedAnswer, sortAnswers) match {
+      case Some(errorMessage) => fail(errorMessage)
+      case None =>
+    }
+  }
+
+  protected def dataFrameToSeqScanNode(df: DataFrame): SeqScanNode = {
+    new SeqScanNode(
+      df.queryExecution.sparkPlan.output,
+      df.queryExecution.toRdd.map(_.copy()).collect())
+  }
+
+}
+
+/**
+ * Helper methods for writing tests of individual local physical operators.
+ */
+object LocalNodeTest {
+
+  /**
+   * Runs the `LocalNode`s and makes sure the answer matches the expected result.
+   * @param input the input data to be used.
+   * @param nodeFunction a function which accepts the input `LocalNode`s and uses them to
+   *                     instantiate the local physical operator that's being tested.
+   * @param expectedAnswer the expected result in a [[Seq]] of [[Row]]s.
+   * @param sortAnswers if true, the answers will be sorted by their toString representations prior
+   *                    to being compared.
+   */
+  def checkAnswer(
+    input: Seq[SeqScanNode],
+    nodeFunction: Seq[LocalNode] => LocalNode,
+    expectedAnswer: Seq[Row],
+    sortAnswers: Boolean): Option[String] = {
+
+    val outputNode = nodeFunction(input)
+
+    val outputResult: Seq[Row] = try {
+      outputNode.collect()
+    } catch {
+      case NonFatal(e) =>
+        val errorMessage =
+          s"""
+              | Exception thrown while executing local plan:
+              | $outputNode
+              | == Exception ==
+              | $e
+              | ${org.apache.spark.sql.catalyst.util.stackTraceToString(e)}
+          """.stripMargin
+        return Some(errorMessage)
+    }
+
+    compareAnswers(outputResult, expectedAnswer, sortAnswers).map { errorMessage =>
+      s"""
+          | Results do not match for local plan:
+          | $outputNode
+          | $errorMessage
+       """.stripMargin
+    }
+  }
+
+  private def compareAnswers(
```

Comment:
I added `compareAnswers` to object `SQLTestUtils` and removed it from SparkPlanTest and LocalNodeTest.